### ISSUE TITLE
Missing dependency for drag and drop

### ISSUE THREAD

**Issue by jflamy** [vaadin/flow#16788]
### Description of the bug

Attempting to use Grid Drag and Drop.  
Everything compiles fine, application runs fine until when invoking `grid.setRowsDraggable(true);`
At that time, a dependency is fetched using reflection and is not found.
```
Caused by: java.lang.NoClassDefFoundError: com/vaadin/flow/component/dnd/DragSource
	at com.vaadin.flow.component.grid.Grid.setRowsDraggable(Grid.java:4257)
```

Project is configured using maven

```
	<dependencyManagement>
		<dependencies>
			<dependency>
				<groupId>com.vaadin</groupId>
				<artifactId>vaadin-bom</artifactId>
				<version>${vaadin.version}</version>
				<type>pom</type>
				<scope>import</scope>
			</dependency>
```

`vaadin.version` is 23.3.12

### Expected behavior

Drag and drop should not cause an error.

### Minimal reproducible example

Basically copied the code from the Vaadin Cookbook.

### Versions

Vaadin: 23.3.12
Flow: 23.3.9
Java: JetBrains s.r.o. 17.0.6
OS: amd64 Windows 11 10.0
Browser: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36 Edg/113.0.1774.42
Live reload: Java active (HotswapAgent): Front end active

**Comment by jflamy** [comment]
Sorry. Driver error.  Exclusion was present.
